src/Catalog/Registration/IPackagePathProvider.cs

```diff
@@ -4,10 +4,11 @@
 namespace NuGet.Services.Metadata.Catalog.Registration
 {
     /// <summary>
-    /// An interface for providing the relative path of a package in the base address
+    /// An interface for providing the relative path of a package or an icon in the base address
     /// </summary>
     public interface IPackagePathProvider
     {
+        string GetIconPath(string id, string version);
```

Comment:
Is that better that we have another interface and implementation to get icon path? It will enhance single-responsibility principle.

Reply:
That would probably be nice if we had proper DI in V3 jobs. Right now it will cause too much noise. Currently they are logically grouped by container type (packages vs flat container), I can drop the "Packages" work from the interface name, but icon path is a package property, so current implementation seems logical for me.

Reply:
Ok